source/common/modules/markdown-editor/plugins/render-math.ts

```diff
@@ -77,7 +74,21 @@ commands.markdownRenderMath = function (cm: CodeMirror.Editor) {
     // Enable on-click closing of rendered Math elements.
     mathSpan.onclick = (e) => { textMarker.clear() }
 
-    katex.render(myMarker.eq, mathSpan, { throwOnError: false, displayMode: myMarker.displayMode })
+    // Create list of custom commands
+    const commands = await ipcRenderer.invoke('assets-provider', { command: 'getLatexCommands' }) as LatexCommand[]
+    const macros = commands
+      .filter(command => command.definition !== undefined)
+      .reduce<Record<string, string>>((map, command) => {
+      map['\\' + command.name] = command.definition ?? ''
+      return map
+    }, {})
+
+    // Render the equation
+    katex.render(myMarker.eq, mathSpan, {
+      throwOnError: false,
+      displayMode: myMarker.displayMode,
+      macros: macros
```

Comment:
Yes, looking at this code, you definitely should not be building this list here. Rather, you could hook to `window.ipc.on` and update some module-global macros variable whenever there is an update.

Reply:
Okay, but how do I then initialize the module variable (async calls on the module level are not yet supported, or are they?)